wordpress-mobile/WordPress-Android
# Plans in site creation: Remove site domains feature flag

Fixes #19252 

Removing the site domains feature flag as it a stable feature

<img width=320 src="https://github.com/wordpress-mobile/WordPress-Android/assets/990349/7d89184a-c3ea-4a94-9be5-542632c4e8bc" />


To test:

- Launch JP app
- `Go` to Debug settings (Me -> Debug settings)
- `Verify` `site_domains_remote_field` as shown above is not present
- `Switch` to  `Menu` tab on `My Site` 
- `Verify` menu item `Domains` is present under Manage
- `Tap` on `Domains`
- `Verify` it opens domains screen

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Updated unit tests

PR submission checklist:
